### DISCUSSION TITLE
API and WEBUI have different effects

### DISCUSSION THREAD

**Discussion by xiaoxuwei12345** [Mikubill/sd-webui-controlnet#1482] [General]
The image generated by the API method of tile is different from that generated by the WEBUI method. As shown in the figure, the image generated by the API method is blue. Is my parameter setting incorrect?
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/117269372/bd033171-85c2-48c6-9301-8cbb46ea1a03)

    "alwayson_scripts": {
        "controlnet": {
            "args": [
                {
                    "input_image": encoded_image2,
                    "module": "tile_resample",
                    "model": "control_v11f1e_sd15_tile [a371b31b]",
                    "weight": 1,
                    "resize_mode": "Scale to Fit (Inner Fit)",
                    "lowvram": False,
                    "processor_res": 512,
                    "threshold_a": 0,
                    "threshold_b": 64,
                    "guidance_start": 0,
                    "guidance_end": 1,
                    "down_sampling_rate": 1,
                    "control_mode": 0,
                }
            ]
        }
    }

**Comment by kft334** [comment]
threshold_a is down_sampling_rate. The threshold arguments are used for ALL arguments that are not common to all(most) modules as far as I know.

**Reply by xiaoxuwei12345** [reply]
Hi, Mr. kft334
Thank you!
Please see the picture above. Do you know why tile's API and WEBUI have different effects? I tested canny, and the effect of canny's API is the same as that of WEBUI.

**Comment by missionfloyd** [comment]
The [example](https://github.com/Mikubill/sd-webui-controlnet/wiki/API#integrating-sdapiv12img) on the wiki is wrong.
```python
# Read Image in RGB order
img = cv2.imread('your_image.jpg')[:, :, ::-1]

# Encode into PNG and send to ControlNet
retval, bytes = cv2.imencode('.png', img)
```
`imread()` returns an array in BGR order.
`[:, :, ::-1]` reverses it to RGB.
`imencode()` expects BGR and reverses it again, resulting in a BGR image.

![out](https://github.com/Mikubill/sd-webui-controlnet/assets/4073789/473b6608-b4db-48c5-b99f-46c3ff6747e5)

Removing `[:, :, ::-1]` fixes it.
![out](https://github.com/Mikubill/sd-webui-controlnet/assets/4073789/1a8dd535-eb09-4f22-ab0d-e498bb68c2b7)

**Reply by lllyasviel** [reply]
It seems ture. will fix

**Reply by lllyasviel** [reply]
fixed i think
